modules/http-api/src/main/java/org/mule/runtime/http/api/client/HttpClient.java

```diff
@@ -62,8 +89,37 @@ HttpResponse send(HttpRequest request, int responseTimeout, boolean followRedire
    * @param followRedirects whether or not to follow redirect responses
    * @param authentication the optional {@link HttpAuthentication} to use
    * @return a {@link CompletableFuture} that will complete once the {@link HttpResponse} is available
+   * @deprecated use {@link #sendAsync(HttpRequest, HttpRequestOptions)} instead
+   */
+  @Deprecated
+  default CompletableFuture<HttpResponse> sendAsync(HttpRequest request, int responseTimeout, boolean followRedirects,
+                                                    HttpAuthentication authentication) {
+    return sendAsync(request, new DefaultHttpRequestOptions(responseTimeout, followRedirects, authentication));
+  }
+
+  /**
+   * Same as {@link #sendAsync(HttpRequest, HttpRequestOptions)} but using default options.
+   *
+   * @param request the {@link HttpRequest} to send
+   * @return a {@link CompletableFuture} that will complete once the {@link HttpResponse} is available
+   */
+  default CompletableFuture<HttpResponse> sendAsync(HttpRequest request) {
+    return sendAsync(request, HttpRequestOptions.builder().build());
+  }
+
+  /**
+   * Sends a HttpRequest without blocking the current thread. When a response is available or the request times out the returned
+   * {@link CompletableFuture} will be completed. Be aware that the response body processing will be deferred so that the response can
+   * be processed even when a large body is still being received. If the full response is needed right away then the provided
+   * {@link HttpResponse} must be read in a different thread so that it does not block the {@link HttpClient} threads handling the
+   * response. It's therefore recommended to use {@link CompletableFuture#get()} or any of the async methods available, such as
+   * {@link CompletableFuture#whenCompleteAsync(BiConsumer, Executor)}, to handle the response is those scenarios since they guarantee
+   * executing on a different thread.
+   *
+   * @param request the {@link HttpRequest} to send
+   * @param options the {@link HttpRequestOptions} to use
+   * @return a {@link CompletableFuture} that will complete once the {@link HttpResponse} is available
    */
-  CompletableFuture<HttpResponse> sendAsync(HttpRequest request, int responseTimeout, boolean followRedirects,
-                                            HttpAuthentication authentication);
+  CompletableFuture<HttpResponse> sendAsync(HttpRequest request, HttpRequestOptions options);
```

Comment:
add a default implementation that calls the previous `sendAsync` method, to keep apis compatible.